File: MaximumSubarray.java
public class MaximumSubarray {
    public static void main(String[] args) {
        int[] nums={-2,1,-3,4,-1,2,1,-5,4};
        System.out.println(maxSubArray(nums));
    }
    public static int maxSubArray(int[] nums){
        int max_sum=nums[0];
        int current_sum=0;

        for (int i = 0; i < nums.length; i++) {
            current_sum+=nums[i];

            if(current_sum>max_sum){
                max_sum=current_sum;
            }
            if(current_sum < 0){
                current_sum=0;
            }
            
    }
    return max_sum;
    }
}

File: maximum_subarray.py
def maxSubArray(nums:list[int])->int:
    max_sum=nums[0]
    current_sum=0
    
    
    if len(nums)==1:
        current_sum=max_sum
        
        return max_sum
    
    for i in range(len(nums)):
        current_sum+=nums[i]
        if current_sum>max_sum:
            max_sum=current_sum
        if current_sum<0:
            current_sum=0
            
    return max_sum


nums=[-2,1,-3,4,-1,2,1,-5,4]
ans=maxSubArray(nums)
print(ans)    
    